src/main/java/uk/nhs/tis/trainee/usermanagement/service/EventPublishService.java

```diff
@@ -54,6 +66,26 @@ public void publishSingleProfileSyncEvent(String traineeTisId) {
     log.info("Sending single profile sync event for trainee id '{}'", traineeTisId);
 
     DataRequestEvent dataRequestEvent = new DataRequestEvent("Person", traineeTisId);
-    messagingTemplate.convertAndSend(queueUrl, dataRequestEvent);
+    queueMessagingTemplate.convertAndSend(queueUrl, dataRequestEvent);
+  }
+
+  /**
+   * Public an event containing details of a user account email changing.
+   *
+   * @param userId        The ID of the user account.
+   * @param traineeId     The ID of the trainee.
+   * @param previousEmail The original email associated with the account.
+   * @param newEmail      The new email associated with the account.
+   */
+  public void publishEmailUpdateEvent(String userId, String traineeId, String previousEmail,
+      String newEmail) {
+    log.info("Publishing email update event for previous email '{}' and new email '{}'.",
+        previousEmail, newEmail);
+    EmailUpdateEvent event = new EmailUpdateEvent(userId, traineeId, previousEmail, newEmail);
+    notificationMessagingTemplate.convertAndSend(userAccountUpdateTopicArn, event, Map.of(
+        NOTIFICATION_SUBJECT_HEADER, "Account Email Updated",
+        MESSAGE_GROUP_ID_HEADER, userId,
+        "producer", "tis-trainee-user-management"
```

Comment:
This is interesting. Are you keen to see anything published to SNS including this sort of information so we know where it originally came from? I haven't noticed us using these headers before.

Reply:
It's a bit of a trial off the back of my realisation it's not possible to identify the queue a DLQ message came from. If we can at least identify where it originated we may have a better chance.